base/stream.jl

```diff
@@ -926,12 +934,25 @@ end
 
 uv_write(s::LibuvStream, p::Vector{UInt8}) = uv_write(s, pointer(p), UInt(length(p)))
 function uv_write(s::LibuvStream, p::Ptr, n::UInt)
+    cnt::Int = 0
+    if OS_NAME === :Darwin && n > rw_chunk_sz
+        nchunks = div(n, rw_chunk_sz)
+        for i in 1:nchunks
+           cnt = cnt + uv_write2(s, p+cnt, UInt(rw_chunk_sz))
+        end
+        cnt = cnt + uv_write2(s, p+cnt, UInt(rem(n, rw_chunk_sz)))
+    else
+        cnt = uv_write2(s, p, n)
+    end
+    return cnt
+end
+function uv_write2(s::LibuvStream, p::Ptr, n::UInt)
     check_open(s)
     uvw = Libc.malloc(_sizeof_uv_write)
     uv_req_set_data(uvw,C_NULL)
     err = ccall(:jl_uv_write,
                 Int32,
```

Comment:
`jl_uv_write` should at least be fixed to return `Cssize_t`, the `libuv` library is broken, in that it returns 32-bit `int` instead of `ssize_t` from various functions like `uv_write`, where it states that the function is supposed to be equivalent to the `pwritev` call.